### ISSUE TITLE
Queries are slow in code but fast in tools

### ISSUE THREAD

**Issue by neohuruyi** [mymarilyn/clickhouse-driver#377]
A query returns 2000000 pieces of data,
 taking 33 seconds in the code and 1 second in the DBeaver Tool
# Full Code

```
from clickhouse_driver import Client, connect
from clickhouse_driver.dbapi.extras import DictCursor
ch_pool = connect(host=host, port=port, database=db_name, user=user, password=password)
cur = ch_pool.cursor(cursor_factory=DictCursor)
sql = """
        SELECT
        statistic_date,
        group_id,
        group_name ,
        merchant_id ,
        merchant_name ,
        panel_id ,
        member_card,
        member_id,
        order_no,
        order_status,
        quantity ,
        price,
        total_price ,
        discounts_price ,
        pay_time ,
        time_period ,
        promotion_name ,
        coupon_name
    FROM
        rev_order_detail rod
    WHERE
        group_id = 2
        AND source = 0
        AND statistic_date BETWEEN '2022-03-08' AND '2023-05-08'
    """
cur.execute(sql)
data = cur.fetchall()
```
# and then 

It's just an ordinary sql
So I am not sure what the problem is

**Comment by neohuruyi** [comment]
> Does the same query return all the rows without `LIMIT` in DBeaver?

yes,the same query return all the rows without LIMIT in DBeaver

**Comment by konnectr** [comment]
> > Does the same query return all the rows without `LIMIT` in DBeaver?
> 
> yes,the same query return all the rows without LIMIT in DBeaver

Dbeaver sets the limit automatically. Try run query in clickhouse-client

**Comment by neohuruyi** [comment]
> > > Does the same query return all the rows without `LIMIT` in DBeaver?
> > 
> > 
> > yes,the same query return all the rows without LIMIT in DBeaver
> 
> Dbeaver sets the limit automatically. Try run query in clickhouse-client

**Comment by neohuruyi** [comment]
> > > Does the same query return all the rows without `LIMIT` in DBeaver?
> > 
> > 
> > yes,the same query return all the rows without LIMIT in DBeaver
> 
> Dbeaver sets the limit automatically. Try run query in clickhouse-client

But I tried to add restrictions， the  sql + 'LIMIT 1, 2000000',  The query speed in the Dbeaver  tool is still very fast

**Comment by neohuruyi** [comment]
if query return one hundred thousand rows in code is fast,  More than one million queries will slow down

**Comment by xzkostyan** [comment]
What speed do you expect from clickhouse-driver and why? I don't see any particular timings and limits in your last comment.

How should I reproduce the problem?

**Comment by neohuruyi** [comment]
> What speed do you expect from clickhouse-driver and why? I don't see any particular timings and limits in your last comment.
> 
> How should I reproduce the problem?

I expect the time to be within 10 seconds. What I mean is that the issue of adding 'limit 1, 2000000' after the problem still exists. Alternatively, you can try querying two million rows to test the query time

**Comment by neohuruyi** [comment]
> What speed do you expect from clickhouse-driver and why? I don't see any particular timings and limits in your last comment.
> 
> How should I reproduce the problem?
I find Dbeaver sets the limit automatically，but I still want  to konw  How to quickly obtain a large amount of data at once

**Comment by neohuruyi** [comment]
> > > Does the same query return all the rows without `LIMIT` in DBeaver?
> > 
> > 
> > yes,the same query return all the rows without LIMIT in DBeaver
> 
> Dbeaver sets the limit automatically. Try run query in clickhouse-client

You are right, querying a large amount of data on clickhouse client is also slow
